cunumeric/_ufunc/ufunc.py

```diff
@@ -561,7 +561,7 @@ def _find_common_type(
         scalar_types = []
         array_types = []
         for arr, orig_arg in zip(arrs, orig_args):
-            if arr.ndim == 0:
+            if arr.ndim == 0 and not isinstance(orig_arg, ndarray):
```

Comment:
Arg, I guess this was one of those places changed earlier.  This is OK, you do lose the value inspection part of course (not that it is often good or even desirable).

I don't have a great solution to offer that would fully compatible (besides converting to a NumPy array right now).  This is a very mild divergence that should be future proof, though.

> Or we could depart from NumPy's semantics altogether, and never consult the actual values of scalars when picking types.

I am in the process of changing this in NumPy as well, to make it basically so that only Python `float`, `int`, and `complex` have special handling (also that is proving a bit tricky.